tsconfig.json

```diff
@@ -17,9 +17,7 @@
     "skipLibCheck": true,
     "tsBuildInfoFile": ".tsbuildinfo",
     "lib": ["es6", "dom", "es2017"],
-    "incremental": true,
-    "declaration": true,
-    "emitDeclarationOnly": true // to emit at least something until we can update Typescript to v4 then we will want to use `--noEmit`
```

Comment:
I moved these back out because the linter used this too and I didn't want any adverse side effects with it, either

Reply:
💬 Feels weird that we need an own configuration file for type checks 🤷

Reply:
I think it's the github action that needed `emitDeclarationOnly`. So, we need a config file in order to do type checking with a github action, maybe?